### PR TITLE
RDCC-5671 Common: Change in the LoV API to return the generic category details when a Service ID is passed as an input parameter

### PR DESCRIPTION
**Pull request by vilasshelke-hmcts**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RDCC-5671

### Change description ###

Common: Change in the LoV API to return the generic category details when a Service ID is passed as an input parameter

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/functionalTest/java/uk/gov/hmcts/reform/cdapi/CommonDataApiFunctionalTest.java [hunk 1]
```diff
@@ -304,5 +310,39 @@ void shouldReturnSuccessWithServiceIdNoChilds() {
...
+            );
+        if (OK.value() == response.getStatusCode()) {
+            var categories = response.getBody().as(Categories.class);
+            assertNotNull(categories);
+            assertThat(categories.getListOfCategory()).hasSizeGreaterThan(1);
+            categories.getListOfCategory().forEach(h -> assertEquals("ListingStatus", h.getCategoryKey()));
+        } else {
+            assertEquals(NOT_FOUND.value(), response.getStatusCode());
+        }
+    }
```
Comment (rajkatla-hmcts) [inline src/functionalTest/java/uk/gov/hmcts/reform/cdapi/CommonDataApiFunctionalTest.java:346]: these 2 functional test cases are partially covered, as your test case checking for either response can be NOT_FOUND or OK , can you pls add integration test cases to prove that your change will fetch exact number of records which have empty service id and assert their count and content.  
For this you have to insert couple records with empty service id in db and one record with child records, then call this endpoint with empty service id for that category key. 
Then check one record has child nodes and another one does not have.

**Review by rajkatla-hmcts — CHANGES_REQUESTED** [review]

File: src/functionalTest/java/uk/gov/hmcts/reform/cdapi/CommonDataApiFunctionalTest.java (hunk 1 above)
Comment (vilasshelke-hmcts) [reply src/functionalTest/java/uk/gov/hmcts/reform/cdapi/CommonDataApiFunctionalTest.java:346]: 1. these 2 functional test cases are partially covered, as your test case checking for either response can be NOT_FOUND or OK, can you pls add integration test cases to prove that your change will fetch exact number of records which have empty service id and assert their count and content- 
@rajkatla-hmcts : Is data dynamic or permanent?

2. For this you have to insert couple records with empty service id in db and one record with child records, then call this endpoint with empty service id for that category key. - 
@rajkatla-hmcts - This is covered with categoryKey = ListingStatus

3. Then check one record has child nodes and another one does not have.
@rajkatla-hmcts - I did not do any changes in child node functionality.

**Review by rajkatla-hmcts — APPROVED** [review]

**Review by kavithagade12 — APPROVED** [review]
